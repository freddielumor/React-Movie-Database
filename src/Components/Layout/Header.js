import React from 'react';
import { Link } from 'react-router-dom';
import {
    Grid,
    Row,
    Col,
    Nav,
    Icon,
    Divider
} from 'rsuite';
import './Header.scss';

// Components
import Search from '../../Components/Search/Search';


const Header = () => (
    <header className="app__header">
        <Grid>
            <Row>
                <Col xs={24}>
                    <h1><a className="app__header--headline" href="/"> React Movie Database</a></h1>
                    <p>Made by <a className="app__footer--freddie" href="https://github.com/freddielumor" target="_blank">Freddie</a></p>
                    <Divider className="app__divider" />
                    <Search />
                    {/* <Nav>
                        <Nav.Item
                            icon={<Icon icon="home" />}
                            componentClass={Link}
                            to="/"
                        >
                            Home
                    </Nav.Item>
                        <Nav.Item
                            componentClass={Link}
                            to="/new-releases"
                        >
                            New
                    </Nav.Item>
                        <Nav.Item
                            componentClass={Link}
                            to="/categories"
                        >
                            Categories
                    </Nav.Item>
                    </Nav> */}
                </Col>
            </Row>
        </Grid>

    </header>
);

export default Header;