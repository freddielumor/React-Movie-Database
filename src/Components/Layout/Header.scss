.app__header {

    .headline {
        &:hover {}
    }

    .app__divider {
        margin-bottom: 0;
    }
}