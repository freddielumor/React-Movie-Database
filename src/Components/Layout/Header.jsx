import React from "react";
import { Grid, Row, Col, Divider } from "rsuite";
import "./Header.scss";

const Header = () => (
  <header className="app__header">
    <Grid>
      <Row>
        <Col xs={24}>
          <h1>
            <a
              className="app__header--headline"
              href="/"
              data-testid="appHeader"
            >
              React Movie Database
            </a>
          </h1>
          <p data-testid="appAuthor">
            Made by{" "}
            <a
              className="app__header--freddie"
              href="https://github.com/freddielumor"
              target="_blank"
              rel="noopener noreferrer"
            >
              Freddie
            </a>
          </p>
          <Divider className="app__divider" />
        </Col>
      </Row>
    </Grid>
  </header>
);

export default Header;
