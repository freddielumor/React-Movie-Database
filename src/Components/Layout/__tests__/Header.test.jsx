import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";
import Header from "../Header.jsx";

afterEach(cleanup);

describe("App Header", () => {
  const { asFragment } = render(<Header />);

  // Snapshot
  it("matches snapshot", () => {
    expect(asFragment()).toMatchSnapshot();
  });

  // Unit tests
  it("Displays a title", () => {
    const { getByTestId } = render(<Header />);
    const title = "React Movie Database";
    const appTitle = getByTestId("appHeader");
    expect(appTitle.innerHTML).toBe(title);
  });

  // it("Displays the author", () => {
  //   const { debug, getByText } = render(<Header />);
  //   const author = "Made by Freddie";
  //   const appAuthor = getByText(author);
  //   debug();
  //   expect(appAuthor.innerText).toBe(author);
  // });
});
