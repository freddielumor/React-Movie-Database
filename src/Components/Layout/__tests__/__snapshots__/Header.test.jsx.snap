// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Header matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="app__header"
  >
    <div
      class="rs-grid-container"
    >
      <div
        class="rs-row"
      >
        <div
          class="rs-col rs-col-xs-24"
        >
          <h1>
            <a
              class="app__header--headline"
              data-testid="appHeader"
              href="/"
            >
              React Movie Database
            </a>
          </h1>
          <p
            data-testid="appAuthor"
          >
            Made by 
            <a
              class="app__header--freddie"
              href="https://github.com/freddielumor"
              rel="noopener noreferrer"
              target="_blank"
            >
              Freddie
            </a>
          </p>
          <div
            class="rs-divider app__divider rs-divider-horizontal"
          />
        </div>
      </div>
    </div>
  </header>
</DocumentFragment>
`;
